### ISSUE TITLE
Nucleo board's 0.96 oled screen problem

### ISSUE THREAD

**Issue by kwonkwonjae** [OpenSourceHelpCommunity/Questions#287]
i have an issue with my oled screen printing some garbage dots.
i tried oled.display(), changing my codes several times, but i cant find the answer.
also when i use oled screen's functions, it seems that my nucleo board reacts a bit later when i press my button. is the oled functions creating a block?

#include "mbed.h"
#include "Adafruit_SSD1306.h"
I2C i2c(I2C_SDA, I2C_SCL); // SDA, SC
Adafruit_SSD1306_I2c oled(i2c,D4,0x78,32,128);
InterruptIn sw(USER_BUTTON);
PwmOut motor(D7);
DigitalOut led(LED1);
Ticker ticker;
Ticker ticker2;
volatile bool switch_pressed = false;
float angle = 0;
enum State {SHRINK,UP,EXTEND,DOWN}; State state = SHRINK;
void blink(){led=!led;};
void angleUp(){angle++;};
void angleDown(){angle--;};
void turn(PwmOut &rc,float deg){rc.pulsewidth_us(10*deg + 600);}
void switch_callback() {switch_pressed=true;}

void changeState() {
    switch(state) {
        case SHRINK: // shrink일 때 눌리면 이니까 up의 내용이 있어줘야함
            state = UP;
            oled.printf("GOING UP");
            ticker.attach(blink, 0.1);
            ticker2.attach(&angleUp, 0.01);
            break;
        case UP:
            state = SHRINK;
            oled.printf("SHRINK");
            ticker.detach();ticker2.detach();
            angle = 0;
            led = 0;
            break;
        case EXTEND:
            state = DOWN;
            oled.printf("GOING DOWN");
            ticker.attach(blink,0.1);
            ticker2.attach(&angleDown, 0.01);
            break;
        case DOWN:
            state = EXTEND;
            oled.printf("EXTEND");
            ticker.detach();ticker2.detach();
            angle = 180;
            led = 1;
            break;
    }}

int main() {
    sw.fall(&switch_callback);
    i2c.frequency(400000);
    oled.begin();
    while(1){
        turn(motor,angle);
        if (state == UP){(angle>=180)?(state=DOWN,switch_pressed=1):state=UP;}
        if(state == DOWN){(angle<=0)?(state=UP,switch_pressed=1):state=DOWN;}
        if(switch_pressed){changeState();switch_pressed=false;}
        }
    }
